test/unit/StakingProducts/recalculateEffectiveWeight.js

```diff
@@ -0,0 +1,447 @@
+const { ethers } = require('hardhat');
+const { expect } = require('chai');
+
+const { parseEther } = ethers.utils;
+const { Zero, One } = ethers.constants;
+
+const {
+  allocateCapacity,
+  depositTo,
+  burnStake,
+  setStakedProducts,
+  daysToSeconds,
+  burnStakeParams,
+  buyCoverParamsTemplate,
+  newProductTemplate,
+} = require('./helpers');
+const { setEtherBalance } = require('../../utils/evm');
+const { increaseTime } = require('../../utils').evm;
+
+const DEFAULT_PRODUCTS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20];
+describe('recalculateEffectiveWeight', function () {
+  it('should return 0 for products not found in stakingPool', async function () {
+    const { stakingProducts } = this;
+    const productIdToAdd = Zero;
+
+    await setStakedProducts.call(this, { productIds: [productIdToAdd] });
+
+    const stakedProduct = await stakingProducts.getProduct(this.poolId, productIdToAdd);
+    expect(stakedProduct.lastEffectiveWeight).to.be.equal(100);
+    expect(stakedProduct.targetWeight).to.be.equal(100);
+
+    const unknownProduct = await stakingProducts.getProduct(this.poolId, productIdToAdd.add(1));
```

Comment:
i'd use a named constant for 

productIdToAdd.add(1)

as the `unknownProductId`